### PR TITLE
Track total bytes and bytes-per-second for UDP sockets

### PR DESCRIPTION
**Pull request by abachman**

### CODE REVIEW

File: F53OSCSocket.h [hunk 1]
```diff
@@ -33,6 +33,27 @@
...
+
+@property (strong) NSMutableArray *history;
+
+- (double) bytes;
+- (void) addBytes:(double)bytes;
+
+- (void) stop;
+
+@end
+
```
Comment (abachman) [inline F53OSCSocket.h]: not sure this is the appropriate way of doing instance variables for non-ARC code.

File: F53OSCSocket.m [hunk 2]
```diff
@@ -126,15 +187,18 @@ - (BOOL) startListening
 {
     if ( _tcpSocket )
         return [_tcpSocket acceptOnPort:_port error:nil];
-    
+
     if ( _udpSocket )
     {
         if ( [_udpSocket bindToPort:_port error:nil] )
+        {
+            _stats = [[[F53OSCStats alloc] init] autorelease];
```
Comment (ccahoon) [inline F53OSCSocket.m]: I may be wrong, but I think you may not want to autorelease here. Who should own this object?

File: F53OSCSocket.h (hunk 1 above)
Comment (ccahoon) [inline F53OSCSocket.h]: In non-arc I think it should be @property (retain) instead of @property (strong)

Comment (abachman) [inline F53OSCSocket.h]: that appears to be correct, updated accordingly

File: F53OSCSocket.m (hunk 2 above)
Comment (abachman) [inline F53OSCSocket.m]: see commit 0a6b9ba

File: F53OSCSocket.m [hunk 3]
```diff
@@ -142,9 +223,14 @@ - (void) stopListening
 {
     if ( _tcpSocket )
         [_tcpSocket disconnectAfterWriting];
-    
+
     if ( _udpSocket )
+    {
         [_udpSocket close];
+        [_stats stop];
```
Comment (ChrisAshworth) [inline F53OSCSocket.m]: We might want to maintain the stats over the whole life of the server, but if so we'll need to find a different place to stop/release the stats object.

**Pull request merged**